src/main/java/com/bootcamp/demo/restapi/BookingsController.java

```diff
@@ -0,0 +1,108 @@
+            LOGGER.error("Failed to create booking with id = " + booking.getId(), e.getMessage());
+            return FAILURE_RESPONSE;
+        }
+    }
+
+    @GetMapping("/getBookingsByUserId/{userId}")
+    public ResponseEntity<LinkedHashSet<Booking>> getBookingsByUserId(@PathVariable(value = "userId") String userId) {
+        LinkedHashSet<Booking> bookingsByUserId = new LinkedHashSet<>();
+        try {
+            bookingsByUserId = bookingService.getBookings(userId);
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.error("Failed to find bookings by user id = " + userId, e.getMessage());
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/getBooking/{id}")
+    public ResponseEntity<Object> getBooking(@PathVariable(value = "id") final String id) {
+        Booking booking = new Booking();
+        try {
+            booking = bookingService.getBookingByID(id);
+            return new ResponseEntity<>(booking, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.error("Failed to get bookings by id = " + id, e.getMessage());
+            return new ResponseEntity<>(booking, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/showAllBookings")
+    public ResponseEntity<LinkedHashSet<Booking>> getAllBookings() {
+        LinkedHashSet<Booking> bookings = new LinkedHashSet<>();
+        try {
+            bookings = bookingService.getAllBookings();
+            return new ResponseEntity<>(bookings, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error("Failed to find all bookings. {}", e.getMessage());
+            return new ResponseEntity<>(bookings, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PostMapping("/bookScooter")
+    @ResponseBody
+    public ResponseEntity<Scooter> bookScooter(final Double lat, final Double longitude) throws ExecutionException, InterruptedException {
+       return new ResponseEntity<>(bookingService.bookAScooter(lat, longitude), HttpStatus.OK);
+    }
+
+    @DeleteMapping("/delete")
+    public ResponseEntity<Object> deleteBooking(@RequestParam final String id) {
+        try {
+            bookingService.deleteBooking(id);
+            return SUCCESS_RESPONSE;
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.error("Failed to delete booking with id = " + id, e.getMessage());
+            return FAILURE_RESPONSE;
+        }
+    }
+
+    @PutMapping("/update")
+    public ResponseEntity<Double> updateBooking(@RequestParam final String id, @RequestParam("end")
+            String endDate, PaymentStatus status) {

```

Comment:
Rather than entering payment status here, you could create a PaymentService to imitate a payment action.
For simplicity you can always return successful payment at 1st.
A easy way to imitate and failed payments would be to create a random between a range, let's say 0-10 exclusive, and for a specific interval of values return success for other failure. This would imitate a failure rate of X %.